src/NLog/Conditions/ConditionMethods.cs

```diff
@@ -166,5 +167,43 @@ public static int Length(string text)
         {
             return text.Length;
         }
+
+        /// <summary>
+        /// Indicates whether the specified regular expression finds a match in the specified input string.
+        /// </summary>
+        /// <param name="input">The string to search for a match.</param>
+        /// <param name="pattern">The regular expression pattern to match.</param>
+        /// <param name="options">A string consisting of the desired options for the test. Available options are : 'i' for ignore case, 'm' for multi-line, 's' for single line and 'x' for ignore pattern whitespace.</param>
+        /// <returns>true if the regular expression finds a match; otherwise, false.</returns>
+        [ConditionMethod("regex-matches")]
+        public static bool RegexMatches(string input, string pattern, string options)
+        {
+            RegexOptions regexOpts =
+#if SILVERLIGHT
+                RegexOptions.None;
+#else
+                RegexOptions.Compiled;
+#endif
+
+            if (options.IndexOf('i') != -1)
+            {
+                regexOpts |= RegexOptions.IgnoreCase;
```

Comment:
It would be nice to create a helper for this. (for at least the indexof, but maybe also the `|=`)